### ISSUE TITLE
Unable to authenticate, need: Basic realm="GitHub"

### ISSUE THREAD

**Issue by dptoot** [Cox-Automotive/alks-cli#150]
Hiya!

Trying to install the package from npm and getting an authentication error.  Looks like this has been an issue in the past with repos that have `always-auth` as ours does.

```
npm ERR! code E401
npm ERR! Unable to authenticate, need: Basic realm="GitHub"
```

https://github.com/Cox-Automotive/alks-cli/pull/144
https://github.com/Cox-Automotive/alks-cli/pull/121

absolute dependency still exists in package.json

https://github.com/Cox-Automotive/alks-cli/blob/master/package.json#L44

Thanks!

**Comment by apottere** [comment]
@aaron-seitz no, per my PR this is an issue with using `always-auth=true` (required for private registries that need auth on `GET`s) and having a generic URL as a dependency.  If you use the shorthand for [GitHub URLs](https://docs.npmjs.com/files/package.json#git-urls-as-dependencies), installs work perfectly.

If a GitHub dependency of the form `Cox-Automotive/clortho#1.2.3` is causing issues with the build system, you can try using `git+https://github.com/Cox-Automotive/clortho#1.2.3` to force downloading with https.  Either way is perfectly valid syntax, and the build system should support it.

This has prevented us from updating alks-cli past version `3.0.5`.

**Comment by rsturim** [comment]
I am trying to install the alks-cli on node v13.12.0 (nodejs stable), I am also getting the error message

`Unable to authenticate, need: Basic realm="GitHub"`

Is the alks-cli not compatible with node 13??

**Comment by jrouly** [comment]
@rsturim The error you're seeing is a permissions error, not related to node version compatibility. If you have any custom npm registries defined, try commenting them out temporarily while you install `alks-cli`.

**Comment by amagana3** [comment]
@rsturim we've also seen this issue recently, [specifically here](https://coxauto.slack.com/archives/C4C5MKVQ9/p1601573005024200), where the issue ended up being that there are more than 1 alks instances found. Using the command `which alks` to find if there are more than one can help resolve this issue.

**Comment by ddcrjlalumiere** [comment]
I'm running in to this issue as well. My naive assumption is that one could setup a custom registry and then `npm` would check that first and the standard registry if the package is not foundf. Certainly there must be a solution to allow for both custom and default registry that isn't "swap your `.npmrc` whenever you use `npm`"?

**Comment by bondezbond** [comment]
so I have the same issue. Reading into node docs I suggest creating different profiles and using `npmrc` to swap between company and public profiles. https://docs.npmjs.com/configuring-your-registry-settings-as-an-npm-enterprise-user

**Comment by superseeker13** [comment]
I'm experiencing the same issue with a different repo and it is rather frustrating to swap npmrc files.  There needs to be a better way of doing this.

**Comment by ransomhall** [comment]
Count me in on this issue. By switching to `always-auth=false` in .npmrc, I got past it. Would be nice to not have to remember to do that every time I upgrade alks.

**Comment by americk0** [comment]
This issue should be resolved for all users in versions 3.2.5 and up now that PR #191 has been deployed since we're no longer using github URLs for the clortho dependency. Please open another issue if any other issues occur as a result of this change
